### PR TITLE
Fix test project for CI (Continuous Integration) by Jenkins.

### LINKED ISSUE #172: To fix the test project for CI (Continuous Integration) by Jenkins.
# Requirement

In accordance with the following procedure, set up and construction of jenkins,
- http://techinfoofmicrosofttech.osscons.jp/index.php?Jenkins%E6%A7%8B%E7%AF%89%E6%89%8B%E9%A0%86
- http://opentouryo.osscons.jp/index.php?Jenkins%E6%A7%8B%E7%AF%89

and perform the CI (continuous integration) using the [test projects](https://github.com/OpenTouryoProject/OpenTouryo/tree/develop/root/programs/C%23/Tests).

Along with this, to optimize the test code.
### About path

I have considered that the path of the problems can occur when deploy too deep position.

Therefore, if the plug-in needs to be located in the under the workspace folder.
In this case, this has been addressed in the XCOPY.
see : http://opentouryo.osscons.jp/index.php?Jenkins%E6%A7%8B%E7%AF%89#sffe761b

### PR DESCRIPTION
**Pull request by daisukenishino**
fixed #172

### CODE REVIEW

**Comment by yusukemaegawa** [comment]
@daisukenishino 

Please describe the summary of this pull request.
**What was the issue?**
**And, what did you modify?**

Others can not understand what you want to do.

**Comment by daisukenishino** [comment]
@yusukemaegawa 

I added the description.
https://github.com/OpenTouryoProject/OpenTouryo/pull/168#issue-144770630

File: root/programs/C#/Tests/Public.Test/Public.Test.csproj.user [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'">
+    <StartAction>Program</StartAction>
+    <StartProgram>C:\root\programs\C#\Tests\packages\NUnit.Runners.2.6.3\tools\nunit.exe</StartProgram>
```
Comment (yusukemaegawa) [inline root/programs/C#/Tests/Public.Test/Public.Test.csproj.user]: @daisukenishino 

By default, Jenkins deploys the source code downloaded from GitHub at the following location.

```
C:\Program Files\Jenkins\jobs\[job name]\workspace
```

Is this location (C:\root\...) OK?

**Comment by yusukemaegawa** [comment]
@daisukenishino 

You should use "MSBuild Integration feature of NuGet", and remove packages folder.

File: root/programs/C#/Tests/Public.Test/Public.Test.csproj.user (hunk 1 above)
Comment (daisukenishino) [inline root/programs/C#/Tests/Public.Test/Public.Test.csproj.user]: @yusukemaegawa 

I have considered that the path of the problems can occur when deploy too deep position.

Therefore, if the plug-in needs to be located in the under the workspace folder.
In this case, this has been addressed in the XCOPY.
see : http://opentouryo.osscons.jp/index.php?Jenkins%E6%A7%8B%E7%AF%89#sffe761b

Comment (yusukemaegawa) [inline root/programs/C#/Tests/Public.Test/Public.Test.csproj.user]: @daisukenishino 

I think that the job of Jenkins execute nunit-console.exe.
(not nunit.exe.)
Is this code necessary?

Comment (daisukenishino) [inline root/programs/C#/Tests/Public.Test/Public.Test.csproj.user]: @yusukemaegawa 

> Is this code necessary?

This is not for the "CI" by the jenkins.
This is set for the confirmation by the result of manual execution of nunit.

Comment (yusukemaegawa) [inline root/programs/C#/Tests/Public.Test/Public.Test.csproj.user]: @daisukenishino 

The title of this pull request is "Fix test project **for CI (Continuous Integration) by Jenkins**".
If this file is **not for CI**, why did you add this file?

Comment (daisukenishino) [inline root/programs/C#/Tests/Public.Test/Public.Test.csproj.user]: @yusukemaegawa 

This is "StartProgram" settings.
But, this settings is related indirectly to CI of Jenkins.
Purpose of this change is to confirm that the nunit to work properly.

Comment (yusukemaegawa) [inline root/programs/C#/Tests/Public.Test/Public.Test.csproj.user]: @daisukenishino 

>Purpose of this change is to confirm that the nunit to work properly.

If you want to confirm NUnit, you can run runtests.bat.

**Comment by daisukenishino** [comment]
@yusukemaegawa 

>https://github.com/OpenTouryoProject/OpenTouryo/pull/168#issuecomment-208113834
You should use "MSBuild Integration feature of NuGet", and remove packages folder.

Please help me to do this correspondence.

##### In addition,
when upgrade the nunit version then, modification of the test code is required.
Therefore, there is a need to continue the current version.

File: root/programs/C#/Tests/Public.Test/Public.Test.csproj.user (hunk 1 above)
Comment (daisukenishino) [inline root/programs/C#/Tests/Public.Test/Public.Test.csproj.user]: @yusukemaegawa 

What is the runtests.bat.
What thereby becomes convenient ?

Comment (yusukemaegawa) [inline root/programs/C#/Tests/Public.Test/Public.Test.csproj.user]: @daisukenishino 

>What is the runtests.bat.

see:
https://github.com/OpenTouryoProject/OpenTouryo/blob/fix_test_projects/root/programs/C%23/Tests/Public.Test/bat/runtests.bat

>What thereby becomes convenient in the debug of the test ?

In production environment, Jenkins will execute nunit-console.exe, not nunit.exe.
So, you can test in environment that is close to production environment.

**Comment by yusukemaegawa** [comment]
@daisukenishino 

>>https://github.com/OpenTouryoProject/OpenTouryo/pull/168#issuecomment-208113834
>You should use "MSBuild Integration feature of NuGet", and remove packages folder.
>
>Please help me to do this correspondence.

Correspondence method is same as the one of past issue (https://github.com/OpenTouryoProject/OpenTouryo/issues/19#issuecomment-167192103).

>##### In addition,
>when upgrade the nunit version then, modification of the test code is required.
>Therefore, there is a need to continue the current version.

This is obvious.
What are you concerned about?

**Comment by daisukenishino** [comment]
Closed this pull-request for configuration changes.

**Pull request closed without merging**